### PR TITLE
Resource manager add option to copy data from selection and monitoring tabs

### PR DESCRIPTION
**Pull request by oanaBiancaSchiopu**
In case of the same node/nodeSource/Host selection, do not refresh list so that the data can be copied

### CODE REVIEW

**Comment by fviale** [comment]
Retest this please

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSource.java [hunk 1]
```diff
@@ -118,6 +118,25 @@ public int hashCode() {
         return sourceName != null ? sourceName.hashCode() : 0;
     }
 
+    public boolean hasTheSameParams(Object o) {
+        if (o == null || getClass() != o.getClass())
+            return false;
+        NodeSource nodeSource = (NodeSource) o;
+
+        if (this.getSourceName().equals(nodeSource.getSourceName()) &&
```
Comment (fviale) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSource.java:126]: this big if test should be refactored in a method

boolean checkNodeSourceDetailsEquals(NodeSource other)

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSource.java [hunk 2]
```diff
@@ -433,6 +464,24 @@ public int hashCode() {
                 return nodeUrl != null ? nodeUrl.hashCode() : 0;
             }
 
+            public boolean hasTheSameParams(Object o) {
+                if (o == null || getClass() != o.getClass())
+                    return false;
+                Node node = (Node) o;
+
+                if (this.getNodeUrl().equals(node.getNodeUrl()) && this.getNodeState().equals(node.getNodeState()) &&
```
Comment (fviale) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSource.java:472]: same remark please create a test method for Node

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSource.java (hunk 1 above)
Comment (oanaBiancaSchiopu) [reply rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSource.java:126]: I have renamed the methods. 
For the tests, we agreed that they are not worth it as there would be a lot of code to test every possible scenario

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSource.java [hunk 3]
```diff
@@ -118,6 +118,25 @@ public int hashCode() {
         return sourceName != null ? sourceName.hashCode() : 0;
     }
 
+    public boolean checkNodeSourceDetailsEquals(Object o) {
+        if (o == null || getClass() != o.getClass())
```
Comment (amouhoub) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSource.java:122]: @oanaBiancaSchiopu Please add accolades to  your if, even it is a one-liner.

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSource.java [hunk 4]
```diff
@@ -433,6 +464,24 @@ public int hashCode() {
                 return nodeUrl != null ? nodeUrl.hashCode() : 0;
             }
 
+            public boolean checkNodeDetailsEquals(Object o) {
+                if (o == null || getClass() != o.getClass())
```
Comment (amouhoub) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSource.java:468]: Same thing, always use accolades for embedded code blocks, even if it is one line

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/InfoView.java [hunk 5]
```diff
@@ -247,6 +247,9 @@ public void nodesUpdated(Map<String, NodeSource> nodeSources) {
      * .ow2.proactive_grid_cloud_portal.rm.client.NodeSource.Host.Node)
      */
     public void nodeSelected(Node node) {
+        if (node.checkNodeDetailsEquals(selNode))
```
Comment (amouhoub) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/InfoView.java:250]: Use accolades inside the if

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/InfoView.java [hunk 6]
```diff
@@ -359,6 +362,9 @@ private void updateAdditionalInformationCanvasList(
      */
     public void nodeSourceSelected(NodeSource ns) {
 
+        if (ns.checkNodeSourceDetailsEquals(selNS))
```
Comment (amouhoub) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/InfoView.java:365]: Same thing

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/InfoView.java [hunk 7]
```diff
@@ -406,6 +412,10 @@ public void nodeSourceSelected(NodeSource ns) {
      * .ow2.proactive_grid_cloud_portal.rm.client.NodeSource.Host)
      */
     public void hostSelected(Host h) {
+
+        if (h.checkHostDetailsEquals(selHost))
```
Comment (amouhoub) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/InfoView.java:416]: Same

**Review by amouhoub — CHANGES_REQUESTED** [review]

**Review by fviale — APPROVED** [review]

**Comment by fviale** [comment]
Retest this please

**Review by amouhoub — APPROVED** [review]

**Pull request merged**